gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -177,11 +179,14 @@ public static void populateFlowCompilationFailedEventMessage(Optional<EventSubmi
   }
 
   /**
-   * If it is a scheduled flow (and hence, does not have flowExecutionId in the FlowSpec) and the flow compilation is
-   * successful, retrieve the flowExecutionId from the JobSpec.
+   * If it is a scheduled flow (which does not have flowExecutionId in the FlowSpec) and the flow compilation is
+   * successful, add a flowExecutionId using the optional parameter if it exists otherwise retrieve it from the JobSpec.
    */
   public static void addFlowExecutionIdIfAbsent(Map<String,String> flowMetadata,
-      Dag<JobExecutionPlan> jobExecutionPlanDag) {
+      Optional<String> optionalFlowExecutionId, Dag<JobExecutionPlan> jobExecutionPlanDag) {
```

Comment:
tip: rather than updating various existing invocations to pass `Optional.absent()`, create an overloaded version and centralize all migration there:
```
public static void addFlowExecutionIdIfAbsent(Map<String,String> flowMetadata, Dag<JobExecutionPlan> jobExecutionPlanDag) {
  addFlowExecutionIdIfAbsent(flowMetadata, Optional.absent(), jobExecutionPlanDag);
}
```